backend/cloudwatch.go

```diff
@@ -62,6 +62,7 @@ func (cb *CloudWatchBackend) Collect(r *collector.Result) error {
 
 	svc := cloudwatch.New(sess)
 	metrics := []*cloudwatch.MetricDatum{}
+	duration := r.PollDuration.Milliseconds() / 1000
```

Comment:
It looks like the API has the ability to change the PollDuration dynamically with a response header: https://github.com/buildkite/buildkite-agent-metrics/blob/a0e44e3edbd3f2b3ab356eaa174177753ffeaec8/collector/collector.go#L152-L159
https://github.com/buildkite/buildkite-agent-metrics/blob/a0e44e3edbd3f2b3ab356eaa174177753ffeaec8/main.go#L122-L164

Though the API's source code seems to have it hard-coded to 10s.

This would mean that everyone using the cloudwatch backend would suddenly be using the high resolution storage if this were merged, and they started using it.

I think we should make the high resolution storage optional for this reason. Perhaps we can make the default to use the high resolution storage but allow customers to turn it off (and poll less frequently) if it costs them more than they expected.

I don't think we should increase the hard-coded 10s value in the API. Presumably it is more relevant to other backends.

Reply:
oh weird, we do send an interval along in the user agent, but the API ignores it?